mlflow/openai/utils.py

```diff
@@ -151,6 +151,13 @@ def _validate_model_params(task, model, params):
         )
 
 
+def _validate_params_and_envs(params, envs):
+    """
+    params passed at inference time should override envs.
+    """
+    return {k: v for k, v in envs.items() if k not in params} if params else envs
```

Comment:
I'd rename this to `exclude_keys`. When I see `validate`, I normally expect `assert` or a guard (`if cond: ...`) is included.